### PR TITLE
browser: add intercom integration

### PR DESCRIPTION
**Pull request by MikeShi42**

### CODE REVIEW

File: packages/browser/src/index.ts [hunk 1]
```diff
@@ -78,6 +83,35 @@ class Browser {
...
+
+    if (disableIntercom !== true) {
+      resolveAsyncGlobal('Intercom')
+        .catch()
+        .then(() => {
+          window.Intercom('onShow', () => {
+            const sessionUrl = this.getSessionUrl();
+            if (sessionUrl != null) {
+              const metadata = {
+                hyperdxSessionUrl: this.getSessionUrl(),
```
Comment (wrn14897) [inline packages/browser/src/index.ts:97]: Can we use `sessionUrl` here ?

Comment (MikeShi42) [reply packages/browser/src/index.ts:97]: 🤦 yeah haha

File: packages/browser/src/index.ts [hunk 2]
```diff
@@ -92,6 +126,12 @@ class Browser {
 
     Rum.setGlobalAttributes(attributes);
   }
+
+  getSessionUrl(): string | undefined {
+    return Rum.inited
+      ? `${UI_BASE}/sessions?q=process.tag.rum.sessionId%3A${Rum.getSessionId()}`
```
Comment (wrn14897) [inline packages/browser/src/index.ts:132]: Adding " around session id should make search more performant. Something like 
```
`${UI_BASE}/sessions?q=process.tag.rum.sessionId%3A"${Rum.getSessionId()}"`
```

File: packages/browser/src/utils.ts [hunk 3]
```diff
@@ -0,0 +1,16 @@
+export function resolveAsyncGlobal(name: string): Promise<any> {
```
Comment (wrn14897) [inline packages/browser/src/utils.ts:1]: usually I'd say its better to write a simple unit test for this. but I guess you probably had this tested on local already

File: packages/browser/src/index.ts [hunk 4]
```diff
@@ -78,6 +83,35 @@ class Browser {
         maskTextClass: maskClass,
       });
     }
+
+    const tracer = opentelemetry.trace.getTracer('@hyperdx/browser');
+
+    if (disableIntercom !== true) {
+      resolveAsyncGlobal('Intercom')
+        .catch()
+        .then(() => {
```
Comment (wrn14897) [inline packages/browser/src/index.ts:92]: nit: Can we just grab Intercom instance from `then` ? I was thinking maybe adding a TS generic type to `resolveAsyncGlobal`

Comment (MikeShi42) [reply packages/browser/src/index.ts:92]: That's what I did originally, but I spent a few hours trying to figure out why this code doesn't work (and it's due to trying to use the returned intercom instance).

Afaik Intercom follows a somewhat common pattern of loading a placeholder function under `window.Intercom` while the full script is loading, and then replaces it with the actual function when the intercom script is done loading. This means if you use the intercom reference from the initial page load, you'll be hanging on to a placeholder reference (it'll queue up function calls and flush when the intercom script finally loads), which will no-op one the intercom script is done loading.

File: packages/browser/src/utils.ts (hunk 3 above)
Comment (MikeShi42) [reply packages/browser/src/utils.ts:1]: I did think about it, but setting up a timing-dependent test for this simple function didn't seem worth it.

File: packages/browser/src/index.ts (hunk 4 above)
Comment (wrn14897) [reply packages/browser/src/index.ts:92]: Ah...replacing the reference. I see. Didn't see the comment. Thanks for the explanation

**Review by wrn14897 — APPROVED** [review]

**Pull request merged**